components/ai_chat/core/browser/associated_content_manager.h

```diff
@@ -50,9 +50,12 @@ class AssociatedContentManager : public AssociatedContentDelegate::Observer {
 
   // Adds a content delegate to the list of content delegates.
   // If |notify_updated| is true, the conversation will be notified that the
-  // content has been updated.
-  // If |detach_existing_content| is true, the current content will be detached
-  // and the new content will be set as the only content for this conversation.
+  // content has been updated. You might want to avoid notifying if you're
+  // adding multiple contents at once, or if you're replacing (i.e. via a
+  // RemoveContent followed by an AddContent) and you don't want to tell the
+  // ConversationHandler/Frontend about the intermediate states.
```

Comment:
Actually isn't there something to be said here about how it could be problematic if `notify_updated` is true because it does something destructive and if you are mid-operation it could affect the operation?